src/Aspire.Hosting.Azure/Extensions/AzureEventHubsExtensions.cs

```diff
@@ -0,0 +1,101 @@
+
+            serviceBusNamespace.AssignProperty(p => p.Sku.Name, new Parameter("sku", defaultValue: "Standard"));
+
+            var serviceBusDataOwnerRole = serviceBusNamespace.AssignRole(RoleDefinition.EventHubsDataOwner);
+            serviceBusDataOwnerRole.AssignProperty(p => p.PrincipalType, construct.PrincipalTypeParameter);
+
+            serviceBusNamespace.AddOutput("serviceBusEndpoint", sa => sa.ServiceBusEndpoint);
+
+            var azureResource = (AzureEventHubsResource)construct.Resource;
+            var azureResourceBuilder = builder.CreateResourceBuilder(azureResource);
+            configureResource?.Invoke(azureResourceBuilder, construct, serviceBusNamespace);
+
+            foreach (var hub in azureResource.Hubs)
+            {
+                var hubResource = new EventHub(construct, name: hub.Name, parent: serviceBusNamespace);
+                hub.Configure?.Invoke(azureResourceBuilder, construct, hubResource);
+            }
+            
+        };
+        var resource = new AzureEventHubsResource(name, configureConstruct);
+
+        return builder.AddResource(resource)
+                      // These ambient parameters are only available in development time.
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalId)
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
+                      .WithManifestPublishingCallback(resource.WriteToManifest);
+    }
+
+    /// <summary>
+    /// Adds an Azure Event Hubs hub resource to the application model. This resource requires an <see cref="AzureEventHubsResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="builder">The Azure Event Hubs resource builder.</param>
+    /// <param name="name">The name of the hub.</param>
+    public static IResourceBuilder<AzureEventHubsResource> AddHub(this IResourceBuilder<AzureEventHubsResource> builder, string name)
+    {
+#pragma warning disable CA2252 // This API requires opting into preview features
+        return builder.AddHub(name, (_, _, _) => { });
+#pragma warning restore CA2252 // This API requires opting into preview features
+    }
+
+    /// <summary>
+    /// Adds an Azure Event Hubs hub resource to the application model. This resource requires an <see cref="AzureEventHubsResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="builder">The Azure Event Hubs resource builder.</param>
+    /// <param name="name">The name of the hub.</param>
+    /// <param name="configureQueue">Optional callback to customize the hub.</param>
+    [RequiresPreviewFeatures]
+    public static IResourceBuilder<AzureEventHubsResource> AddHub(this IResourceBuilder<AzureEventHubsResource> builder,
+        string name, Action<IResourceBuilder<AzureEventHubsResource>, ResourceModuleConstruct, EventHub>? configureQueue)
```

Comment:
```suggestion
        string name, Action<IResourceBuilder<AzureEventHubsResource>, ResourceModuleConstruct, EventHub>? configureEventHub)
```

There's no such concept of a queue in Event Hubs.

Reply:
Good catch. Boilerplate copypasta error.